common/src/seg_tree.c

```diff
@@ -448,7 +511,10 @@ seg_tree_iter(struct seg_tree* seg_tree, struct seg_tree_node* start)
 void
 seg_tree_rdlock(struct seg_tree* seg_tree)
 {
-    assert(pthread_rwlock_rdlock(&seg_tree->rwlock) == 0);
+    int rc = pthread_rwlock_rdlock(&seg_tree->rwlock);
+    if (rc) {
+        LOGERR("pthread_rwlock_rdlock() failed - rc=%d", rc);
+    }
```

Comment:
I think the extra logging you added is good, but you should also keep the `assert()` so that it stops execution.  I'm worried you could get seg_tree corruption if the program continues without properly acquiring the lock.

Reply:
In fact, in a separate PR, we may want to consider changing the `asserts()` to an `exit(1)` just in case a user builds with NDEBUG (to disable the asserts()).

Reply:
@tonyhutter @MichaelBrim How about letting these functions return `rc` and fixing callers to handle the error?